src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs

```diff
@@ -164,6 +163,10 @@ await ApplyChangeToWorkspaceAsync(w =>
                 onAfterUpdate: null);
         }).ConfigureAwait(false);
 
+        // Set this value early after solution is created so it is available to Razor.  This will get updated
```

Comment:
You probably want to acquire a batch scope around this -- since setting this is going to require going to the global state, we're going to acquire a lock. If there's a batch scope then we'll do it asynchronously, otherwise this will take a blocking lock and block the thread.